oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -36,39 +36,44 @@
    * @param refreshToken A refresh token resulting from a OAuth2 consent flow.
    */
   public UserCredentials(String clientId, String clientSecret, String refreshToken) {
-    this(clientId, clientSecret, refreshToken, null, null);
+    this(clientId, clientSecret, refreshToken, null, null, null);
   }
 
   /**
-   * Constructor allow override of transport for getting access tokens.
+   * Constructor to allow both refresh token and initial access token for 3LO scenarios.
    *
    * @param clientId Client ID of the credential from the console.
    * @param clientSecret Client ID of the credential from the console.
    * @param refreshToken A refresh token resulting from a OAuth2 consent flow.
-   * @param transport HTTP object used to get acces tokens.
+   * @param accessToken Initial or temporary access token.
    */
   public UserCredentials(
-      String clientId, String clientSecret, String refreshToken, HttpTransport transport) {
-    this(clientId, clientSecret, refreshToken, transport, null);
+      String clientId, String clientSecret, String refreshToken, AccessToken accessToken) {
+    this(clientId, clientSecret, refreshToken, accessToken, null, null);
   }
 
+
   /**
-   * Constructor allowing override of transport and URL for getting access tokens.
+   * Constructor with all parameters allowing custom transport and server URL.
    *
    * @param clientId Client ID of the credential from the console.
    * @param clientSecret Client ID of the credential from the console.
    * @param refreshToken A refresh token resulting from a OAuth2 consent flow.
-   * @param transport HTTP object used to get acces tokens.
-   * @param tokenServerUrl URL of the endpoint that provides tokens.
+   * @param accessToken Initial or temporary access token.
+   * @param transport HTTP object used to get access tokens.
+   * @param tokenServerUrl URL of the end point that provides tokens.
    */
   public UserCredentials(String clientId, String clientSecret, String refreshToken,
-      HttpTransport transport, GenericUrl tokenServerUrl) {
+      AccessToken accessToken, HttpTransport transport, GenericUrl tokenServerUrl) {
+    super(accessToken);
```

Comment:
Adding new parameters in the middle of a parameter list is potentially hazardous, especially when nulls are being thrown around.